portfolio/src/main/webapp/script.js

```diff
@@ -40,190 +40,33 @@ function randomizeImage() {
-        },
-        {
-          featureType: 'water',
-          elementType: 'labels.text.stroke',
-          stylers: [{color: colors.color12}]
-        }
-      ]}
-    );
-
-    var geocoder = new google.maps.Geocoder();
-    function callGeocodeAddress(address, userName, userText) {
-      geocodeAddress(geocoder, map, address, userName, userText);
-      console.log("successfully called callGeocodeAddress");
-    }
-    // Allow for call to nested function
-    createMapFunction.callGeocodeAddress = callGeocodeAddress;
-
-  };
-  document.head.appendChild(script);
+    createMapFunction();
 }
 
-function geocodeAddress(geocoder, resultsMap, address, userName, userText) {
-  geocoder.geocode({'address' : address}, function(results, status) {
-    if (status === 'OK') {
-      resultsMap.setCenter(results[0].geometry.location);
-      var contentStr = '<div class="content-marker">' +
-        '<h2 class="first-heading">' + userName + '</h2>' +
-        '<div class="body-content">' +
-        '<p>'+ address + '</p></br>' +
-        '<p>' + userText + '</p></div>';
-      
-      var infowindow = new google.maps.InfoWindow({
-        content : contentStr
-      });
-
-      var marker = new google.maps.Marker({
-        map : resultsMap,
-        position: results[0].geometry.location,
-      });
-      // If user clicks on marker, an info window will pop up displaying user name and address
-      // If user clicks on a marker with an info window opened, window will close.
-      var clickedMarker = false;
-      marker.addListener('click', function() {
-        console
-        if (clickedMarker == false) {
-          infowindow.open(map, marker);
-          clickedMarker = true;
-        } else {
-          infowindow.close(map, marker);
-          clickedMarker = false;
-        }
-      });
-    } else {
-      alert('Geocode was not successful for the following reason: ' + status);
-    }
-  });
-}
 var currUserEmail = "";
 // Function checks whether user is logged in when body loads
 function checkLogin() {
   fetch('/login')
   .then(response => response.json()) // parses the response as JSON
   .then((userInfo) => { // now we can reference the fields in myObject!
-    const loginWrapperElement = document.getElementById('login-wrapper');
+    const loginWrapperElement = document.getElementById('login-link');
     loginWrapperElement.innerHTML = "";
+    var loginLinkPretext = document.getElementById('login-link-pretext');
     if (userInfo.isUserLoggedIn == true) {
       // If user is logged in, provide a log out URl underneath the comment form
+      currUserEmail = userInfo.currUserEmail;
+      loginLinkPretext.innerText = "Not " + currUserEmail + "? "
       loginWrapperElement.innerText = 'Log out';
       loginWrapperElement.href = userInfo.logoutUrl;
       document.getElementById('user-form').style.display = "block";
-      currUserEmail = userInfo.currUserEmail;
+      commentFunction('5');
```

Comment:
nit: I think the code would be more readable if 
1) commentFunction()'s parameter name is called something like "numberOfCommentsToShow"
2)commentFunction() is invoked with an int instead of a character.

Reply:
Fixed in following commit